x-pack/plugins/maps/common/constants.ts

```diff
@@ -79,6 +81,22 @@ export enum SOURCE_TYPES {
   MVT_SINGLE_LAYER = 'MVT_SINGLE_LAYER',
 }
 
+export enum LICENSED_FEATURES {
+  GEO_SHAPE_AGGS = 'GEO_SHAPE_AGGS',
+}
+
+export interface LicensedFeatureDetail {
+  name: string;
+  license: LicenseType;
+}
+
+export const LICENCED_FEATURES_DETAILS: Record<LICENSED_FEATURES, LicensedFeatureDetail> = {
```

Comment:
Maybe `LICENCED_FEATURES_DETAILS` should live outside of common and closer to plugin.ts where its used. How about creating a license features file that contains `LICENCED_FEATURES_DETAILS`